pkg/proxy/conntrack/conntrack.go

```diff
@@ -49,95 +50,139 @@ type Interface interface {
 	ClearEntriesForPortNAT(dest string, port int, protocol v1.Protocol) error
 }
 
-// execCT implements Interface by execing the conntrack tool
-type execCT struct {
-	execer exec.Interface
+// netlinkHandler allows consuming real and mockable implementation for testing.
+type netlinkHandler interface {
+	ConntrackDeleteFilters(netlink.ConntrackTableType, netlink.InetFamily, ...netlink.CustomConntrackFilter) (uint, error)
+}
+
+// conntracker implements Interface by using netlink APIs.
+type conntracker struct {
+	handler netlinkHandler
 }
 
-var _ Interface = &execCT{}
+var _ Interface = &conntracker{}
 
-func NewExec(execer exec.Interface) Interface {
-	return &execCT{execer: execer}
+func New() Interface {
+	return newConntracker(&netlink.Handle{})
 }
 
-// noConnectionToDelete is the error string returned by conntrack when no matching connections are found
-const noConnectionToDelete = "0 flow entries have been deleted"
+func newConntracker(handler netlinkHandler) Interface {
+	return &conntracker{handler: handler}
+}
 
 func protoStr(proto v1.Protocol) string {
 	return strings.ToLower(string(proto))
 }
 
-func parametersWithFamily(isIPv6 bool, parameters ...string) []string {
+// getNetlinkFamily returns the Netlink IP family constant
+func getNetlinkFamily(isIPv6 bool) netlink.InetFamily {
 	if isIPv6 {
-		parameters = append(parameters, "-f", "ipv6")
+		return unix.AF_INET6
 	}
-	return parameters
+	return unix.AF_INET
 }
 
-// exec executes the conntrack tool using the given parameters
-func (ct *execCT) exec(parameters ...string) error {
-	conntrackPath, err := ct.execer.LookPath("conntrack")
-	if err != nil {
-		return fmt.Errorf("error looking for path of conntrack: %v", err)
+// getProtocolNumber return the Assigned Internet Protocol Number.
+// https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml
+func getProtocolNumber(proto v1.Protocol) uint8 {
+	switch proto {
+	case v1.ProtocolTCP:
+		return unix.IPPROTO_TCP
+	case v1.ProtocolUDP:
+		return unix.IPPROTO_UDP
+	case v1.ProtocolSCTP:
+		return unix.IPPROTO_SCTP
 	}
-	klog.V(4).InfoS("Clearing conntrack entries", "parameters", parameters)
-	output, err := ct.execer.Command(conntrackPath, parameters...).CombinedOutput()
-	if err != nil {
-		return fmt.Errorf("conntrack command returned: %q, error message: %s", string(output), err)
-	}
-	klog.V(4).InfoS("Conntrack entries deleted", "output", string(output))
-	return nil
+	return 0
 }
 
 // ClearEntriesForIP is part of Interface
```

Comment:
```suggestion
// ClearEntriesForIP delete the conntrack entries for connections specified by the IP
```

(just to be consistent with the `ClearEntriesForPort` comment below